### PR TITLE
Obey emit_ir more closely

### LINKED ISSUE #224: [fontc] --emit-ir=false leads to TasksFailed([(Be(Font), "IO failure")])
fontc has an --emit-ir flag which is true by default, however it doesn't seem to be possible to disable it.
I tried --no-emit-r or --emit-ir=false but didn't work. Passing `--emit-ir` does nothing different than not passing one at all, since the default is already true.

Should the default be false maybe? Or maybe it can be true but use clap's ArgAction::SetFalse?

### PR DESCRIPTION
**Pull request by rsheeter**
Add a test for compile w/o emitting ir, fix resulting problems. Fixes #224. Aside: discovered https://github.com/googlefonts/fontations/pull/367 along the way.

### CODE REVIEW

**Comment by anthrotype** [comment]
> fix resulting problems.

it would be enlightening if this PR description listed what these problems actually were, otheriwse the onus is on the reviewer to infer those from the code changes

**Comment by rsheeter** [comment]
> what these problems actually were

Most notably that it failed (#224) if IR was off :)

Generally, places where we assumed the files would exist, for example we always created directories and files that are only required for subsequent incremental compiles and the font aggregation was reading files that don't exist when IR is off.

File: fontbe/src/avar.rs [hunk 1]
```diff
@@ -25,8 +26,14 @@ fn to_segment_map(axis: &Axis) -> SegmentMaps {
     // because, per spec, "If the segment map for a given axis has any value maps,
     // then it must include at least three value maps: -1 to -1, 0 to 0, and 1 to 1"
     // so three value maps *must* produce identity.
+    //
+    // 0 entry avar may confuse some clients so emit a default one
```
Comment (anthrotype) [inline fontbe/src/avar.rs:30]: this change to fontbe::avar isn't really related to --emit-ir flag, is it?

Comment (anthrotype) [reply fontbe/src/avar.rs:30]: i see, it's to fix/workaround https://github.com/googlefonts/fontations/pull/367

**Review by anthrotype — APPROVED** [review]
LGTM! +1

**Pull request merged**